### PR TITLE
[WIP][Feature] yarn application mode support prometheus

### PR DESCRIPTION
**Pull request by ChengJie1053**
yarn application mode support prometheus

1.install  prometheus and prometheus

2.Modify the streampark configuration
streampark.prometheus.enable: true
streampark.promgateway.hostUrl: http://ip:9091

See `prometheus.md` for more information

image

image

image

image

### CODE REVIEW

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala [hunk 1]
```diff
@@ -156,6 +156,27 @@ trait FlinkClientTrait extends Logger {
...
     }
 
+    // set prometheus
+    val enablePrometheus: Boolean =
+      InternalConfigHolder.get(CommonConfig.STREAMPARK_PROMETHEUS_ENABLE)
+    val promgatewayHostUrl: String =
+      InternalConfigHolder.get(CommonConfig.STREAMPARK_PROMGATEWAY_HOST_URL)
+
+    if (
+      submitRequest.executionMode == FlinkExecutionMode.YARN_APPLICATION && enablePrometheus && StringUtils
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala:166]: Why is it only enabled in yarn application mode? Doesn't it work in other modes?

Comment (ChengJie1053) [reply streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/trait/FlinkClientTrait.scala:166]: At that time, only yarn application mode was tested

**Comment by Trgree** [comment]
image
It already supports Prometheus. Like this, you can write metrics to Prometheus.

**Comment by ChengJie1053** [comment]
> image It already supports Prometheus. Like this, you can write metrics to Prometheus.

Ok, thank you

**Pull request closed without merging**